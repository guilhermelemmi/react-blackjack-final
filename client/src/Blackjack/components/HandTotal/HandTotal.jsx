import "./HandTotal.css";

function HandTotal ({count, label}) {
  return (
    <div className="handTotal">
      <div className="count" >{count}</div>
      {label} 
    </div>
  )
}

export default HandTotal;
