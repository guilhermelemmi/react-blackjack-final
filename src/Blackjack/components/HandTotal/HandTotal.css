.count {
  background-color: none;
  background-color: rgba(0, 0, 0, 0.25);
  border-radius: 50%;
  border: 2px solid #fff;
  font-size: 32px;
  height: 60px;
  line-height: 60px;
  margin-right: 10px;
  text-align: center;
  width: 60px;
}

.handTotal {
  align-items: center;
  color: white;
  display: flex;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 24px;
  font-weight: bold;
  min-width: 220px;
}