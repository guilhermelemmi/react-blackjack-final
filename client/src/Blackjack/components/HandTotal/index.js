import HandTotal from "./HandTotal";

export default HandTotal;
