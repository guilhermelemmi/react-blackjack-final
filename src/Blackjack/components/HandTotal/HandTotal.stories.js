import HandTotal from "./HandTotal";

export default {
  component: HandTotal,
};

export const Dealer = {
  render: () => <HandTotal count={19} label="Dealer" />,
};

export const Player = {
  render: () => <HandTotal count={19} label="Player" />,
};
