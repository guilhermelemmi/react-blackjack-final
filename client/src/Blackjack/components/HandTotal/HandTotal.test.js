import { render, screen } from "@testing-library/react";
import HandTotal from "./HandTotal";

describe("HandTotal", () => {
  test("It renders count and label", async () => {
    render(<HandTotal count={13} label="Dealer" />);
    const count = screen.getByText("13");
    const label = screen.getByText("Dealer");
    expect(count).toBeInTheDocument();
    expect(label).toBeInTheDocument();
  });

  test("Dealer's count matches snapshot", async () => {
    const view = render(<HandTotal count={13} label="Dealer" />);
    expect(view.container).toMatchSnapshot();
  });

  test("Player's count matches snapshot", async () => {
    const view = render(<HandTotal count={19} label="Player" />);
    expect(view.container).toMatchSnapshot();
  });
});
