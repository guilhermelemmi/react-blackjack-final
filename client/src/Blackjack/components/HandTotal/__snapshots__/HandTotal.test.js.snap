// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandTotal Dealer's count matches snapshot 1`] = `
<div>
  <div
    class="handTotal"
  >
    <div
      class="count"
    >
      13
    </div>
    Dealer
  </div>
</div>
`;

exports[`HandTotal Player's count matches snapshot 1`] = `
<div>
  <div
    class="handTotal"
  >
    <div
      class="count"
    >
      19
    </div>
    Player
  </div>
</div>
`;
